host compile of hamcorebuilder fails (openwrt sdk)

### Prerequisites

* [x] Can you reproduce?
* [x] Are you running the latest version of SoftEtherVPN?

**SoftEther version:**
**Component:** [Server/hamcorebuilder]
**Operating system:** [Linux]
**Architecture:** [64 bit]

### Description

I try to build the latest version for openwrt, but the Host build of hamcorebuilder fails.
If i explicitly add `HOST_CFLAGS += -DHWCAPS_ANDROID_MIPS_OR_ARM` to our makefile it works.
I just wonder if this is required or some other part is not working, i assumed `cpu_features_macros.h` would take care of all the macros?

```
touch /workdir/openwrt-master/build_dir/hostpkg/softethervpn-server-2018-08-12-7c72e008/.configured
CFLAGS="-O2 -I/workdir/openwrt-master/staging_dir/host/include -I/workdir/openwrt-master/staging_dir/hostpkg/include -I/workdir/openwrt-master/staging_dir/target-mips_24kc_musl/host/include" CPPFLAGS="-I/workdir/openwrt-master/staging_dir/host/include -I/workdir/openwrt-master/staging_dir/hostpkg/include -I/workdir/openwrt-master/staging_dir/target-mips_24kc_musl/host/include" CXXFLAGS="" LDFLAGS="-L/workdir/openwrt-master/staging_dir/host/lib -L/workdir/openwrt-master/staging_dir/hostpkg/lib -L/workdir/openwrt-master/staging_dir/target-mips_24kc_musl/host/lib" make -j1 -C /workdir/openwrt-master/build_dir/hostpkg/softethervpn-server-2018-08-12-7c72e008  hamcorebuilder
make[3]: Entering directory '/workdir/openwrt-master/build_dir/hostpkg/softethervpn-server-2018-08-12-7c72e008'
make[4]: Entering directory '/workdir/openwrt-master/build_dir/hostpkg/softethervpn-server-2018-08-12-7c72e008'
make[5]: Entering directory '/workdir/openwrt-master/build_dir/hostpkg/softethervpn-server-2018-08-12-7c72e008'
make[6]: Entering directory '/workdir/openwrt-master/build_dir/hostpkg/softethervpn-server-2018-08-12-7c72e008'
Scanning dependencies of target cpu_features
make[6]: Leaving directory '/workdir/openwrt-master/build_dir/hostpkg/softethervpn-server-2018-08-12-7c72e008'
make[6]: Entering directory '/workdir/openwrt-master/build_dir/hostpkg/softethervpn-server-2018-08-12-7c72e008'
[  1%] Building C object src/Mayaqua/cpu_features/CMakeFiles/cpu_features.dir/src/linux_features_aggregator.c.o
[  1%] Building C object src/Mayaqua/cpu_features/CMakeFiles/cpu_features.dir/src/cpuid_x86_clang_gcc.c.o
[  2%] Building C object src/Mayaqua/cpu_features/CMakeFiles/cpu_features.dir/src/cpuid_x86_msvc.c.o
[  3%] Building C object src/Mayaqua/cpu_features/CMakeFiles/cpu_features.dir/src/cpuinfo_aarch64.c.o
[  4%] Building C object src/Mayaqua/cpu_features/CMakeFiles/cpu_features.dir/src/cpuinfo_arm.c.o
[  5%] Building C object src/Mayaqua/cpu_features/CMakeFiles/cpu_features.dir/src/cpuinfo_mips.c.o
[  6%] Building C object src/Mayaqua/cpu_features/CMakeFiles/cpu_features.dir/src/cpuinfo_ppc.c.o
[  7%] Building C object src/Mayaqua/cpu_features/CMakeFiles/cpu_features.dir/src/cpuinfo_x86.c.o
[  8%] Building C object src/Mayaqua/cpu_features/CMakeFiles/cpu_features.dir/src/filesystem.c.o
[ 10%] Building C object src/Mayaqua/cpu_features/CMakeFiles/cpu_features.dir/src/hwcaps.c.o
/workdir/openwrt-master/build_dir/hostpkg/softethervpn-server-2018-08-12-7c72e008/src/Mayaqua/cpu_features/src/hwcaps.c: In function 'CpuFeatures_GetHardwareCapabilities':
/workdir/openwrt-master/build_dir/hostpkg/softethervpn-server-2018-08-12-7c72e008/src/Mayaqua/cpu_features/src/hwcaps.c:158:53: error: 'AT_HWCAP2' undeclared (first use in this function)
   capabilities.hwcaps2 = GetHardwareCapabilitiesFor(AT_HWCAP2);
                                                     ^~~~~~~~~
/workdir/openwrt-master/build_dir/hostpkg/softethervpn-server-2018-08-12-7c72e008/src/Mayaqua/cpu_features/src/hwcaps.c:158:53: note: each undeclared identifier is reported only once for each function it appears in
make[6]: *** [src/Mayaqua/cpu_features/CMakeFiles/cpu_features.dir/build.make:180: src/Mayaqua/cpu_features/CMakeFiles/cpu_features.dir/src/hwcaps.c.o] Error 1
make[6]: Leaving directory '/workdir/openwrt-master/build_dir/hostpkg/softethervpn-server-2018-08-12-7c72e008'
make[5]: *** [CMakeFiles/Makefile2:280: src/Mayaqua/cpu_features/CMakeFiles/cpu_features.dir/all] Error 2
make[5]: Leaving directory '/workdir/openwrt-master/build_dir/hostpkg/softethervpn-server-2018-08-12-7c72e008'
make[4]: *** [CMakeFiles/Makefile2:386: src/hamcorebuilder/CMakeFiles/hamcorebuilder.dir/rule] Error 2
make[4]: Leaving directory '/workdir/openwrt-master/build_dir/hostpkg/softethervpn-server-2018-08-12-7c72e008'
make[3]: *** [Makefile:251: hamcorebuilder] Error 2
make[3]: Leaving directory '/workdir/openwrt-master/build_dir/hostpkg/softethervpn-server-2018-08-12-7c72e008'
make[2]: *** [Makefile:115: /workdir/openwrt-master/build_dir/hostpkg/softethervpn-server-2018-08-12-7c72e008/.built] Error 2
make[2]: Leaving directory '/workdir/openwrt-master/feeds/packages/net/softethervpn-server'

```

--- a/src/Mayaqua/Encrypt.c
+++ b/src/Mayaqua/Encrypt.c
@@ -4289,11 +4289,9 @@ bool IsAesNiSupported()
 		const Aarch64Features features = GetAarch64Info().features;
 		supported = features.aes;
 	#elif defined(CPU_FEATURES_ARCH_MIPS)
-		const MipsFeatures features = GetMipsInfo().features;
-		supported = features.aes;
+		//const MipsFeatures features = GetMipsInfo().features;  // no features.aes
 	#elif defined(CPU_FEATURES_ARCH_PPC)
-		const PPCFeatures features = GetPPCInfo().features;
-		supported = features.aes;
+		//const PPCFeatures features = GetPPCInfo().features;  // no features.aes
 	#endif
 #endif // _MSC_VER
 

